packages/block-editor/src/utils/dom.js

```diff
@@ -134,41 +134,47 @@ function isScrollable( element ) {
 	);
 }
 
+export const WITH_OVERFLOW_ELEMENT_BLOCKS = [ 'core/navigation' ];
```

Comment:
Not sure if you considered this elsewhere, but I wonder whether in the future we could make this list extensible as an editor setting? That way it can be modified in PHP for custom blocks?

Reply:
Thank you for cherry picking this one! 

@t-hamano had an idea to add it to the block api

I think it's worth investigating if it the issue arises again.